autogluon/utils/tabular/ml/models/lgb/lgb_model.py

```diff
@@ -277,23 +264,27 @@ def hyperparameter_tune(self, X_train, X_test, Y_train, Y_test, scheduler_option
         hpo_model_performances = {}
         for trial in sorted(hpo_results['trial_info'].keys()):
             # TODO: ignore models which were killed early by scheduler (eg. in Hyperband). How to ID these?
-            file_id = "trial_"+str(trial) # unique identifier to files from this trial
-            file_prefix = file_id + "_"
-            trial_model_name = self.name+"_"+file_id
-            trial_model_path = self.path + file_prefix
+            file_id = "trial_" + str(trial)  # unique identifier to files from this trial
+            trial_model_name = self.name + os.path.sep + file_id
+            trial_model_path = self.path_root + trial_model_name + os.path.sep
             hpo_models[trial_model_name] = trial_model_path
             hpo_model_performances[trial_model_name] = hpo_results['trial_info'][trial][scheduler._reward_attr]
 
         logger.log(15, "Time for Gradient Boosting hyperparameter optimization: %s" % str(hpo_results['total_time']))
-        self.params.update(best_hp)
-        # TODO: reload model params from best trial? Do we want to save this under cls.model_file as the "optimal model"
         logger.log(15, "Best hyperparameter configuration for Gradient Boosting Model: ")
         logger.log(15, best_hp)
-        return (hpo_models, hpo_model_performances, hpo_results)
-        # TODO: do final fit here?
-        # args.final_fit = True
-        # final_model = scheduler.run_with_config(best_config)
-        # save(final_model)
+        return hpo_models, hpo_model_performances, hpo_results
+
+    def _get_train_loss_name(self):
```

Comment:
Here I have comment question. As I understand it is for early stopping. Why we are not using the same loss which is used for `fit`  ?

Reply:
This is not for early stopping actually, this is just recorded in reporter for info to the user when they call fit_summary() if I'm not mistaken, although that code was not written by me. The loss metric is often not equivalent to the evaluation metric, since an evaluation metric can often not be directly optimized for. Recording the training evaluation metric would make sense, although we currently don't do it. I personally don't see much benefit in general from observing training metrics, although @jwmueller would know the intent best.

Reply:
Thanks for the explanation, I got the purpose of doing it. But still I have question. Let me reformulate it.  What will happen if for regression I put `eval_metric="r2"` when I run `fit()` ? 
I tried to debug a bit and as far as I see that in this line https://github.com/awslabs/autogluon/blob/b9fb6671c1ab70ddd29e931ef163bee512a86e8c/autogluon/utils/tabular/ml/models/lgb/hyperparameters/lgb_trial.py#L77

the `train_params['params']['objective']` is equal to `regression` which is equivalent to `l2` loss according to LightGBM documentation (https://lightgbm.readthedocs.io/en/latest/Parameters.html#objective). Which means we run `lgb.train()` with loss `l2` rather than `r2`. Is this what we want to do ? So this means that regardless which `eval_metric` I will use for regression problem, we always train the model with `l2` loss. This will work since minimising one loss should minimise the other also, but want to be sure that it is the behaviour which went to have.

Reply:
Yes this code is just to track the performance of the model at each round of iterative training, so the user can inspect the training runs later.

Re the objective mismatch: not every 3rd party model supports all the objectives that autogluon has, so we sometimes will have to make approximations.  However, we are fortunate in this case: minimizing `l2` is equivalent to maximizing `r2`.  This approximation would be less appropriate if we were doing something like quantile regression (say `MAE` loss to estimate median)  and telling `lgb.train()` to still minimize `l2` loss.  So we'd love to have your help if you notice any metric mismatches between what the user specified to autogluon in `eval_metric` and what we pass to each of the models as their objective.